multihash.go

```diff
@@ -227,6 +229,36 @@ func FromB58String(s string) (m Multihash, err error) {
 	return Cast(b)
 }
 
+// FromBinary creates a new multihash from the binary
+// representation. The string is assumed to be a valid multihash.
+// This function will not alloc and will panic if it is unable to
+// parse the string.
+func FromBinary(v string) Multihash {
+	// Sanity check that the string can be parsed.
+	if len(v) < 2 {
+		panic(ErrTooShort)
+	}
+	i := strbinary.UvarintLen(v)
+	digestLen, l := strbinary.Uvarint(v[i:])
+	i += l
+	if len(v[i:]) != int(digestLen) {
+		panic(ErrInvalidMultihash)
+	}
+	return Multihash{v}
+}
+
+// Parts returns the components of the Multihash (Code, Name, Digest)
```

Comment:
Yeah your right since the length is primary useful in the binary representation as a sanity check and to know when the multihash ends in a byte stream.